lib/OpenQA/Schema/Result/Needles.pm

```diff
@@ -83,11 +83,16 @@ sub update_needle {
     my $guard  = $schema->txn_scope_guard;
 
     if (!-f $filename) {
-        $filename = catdir($OpenQA::Utils::prjdir, $filename);
-        if (!-f $filename) {
-            log_error(
-                "Needle file $filename not found where expected. Check $OpenQA::Utils::prjdir for distri symlinks");
+        my $fn = catdir($OpenQA::Utils::prjdir, $filename);
```

Comment:
this is actually wrong - as it will prefer tests/ over share/tests. Which will break our existant needle_dir DB entries :(